src/screens/Feed.js

```diff
@@ -6,65 +6,95 @@ import {
   TouchableWithoutFeedback,
   Text,
   SafeAreaView,
-  Alert
+  Alert,
 } from 'react-native';
 
 import SettingsButton from './app/Components/SettingsButton/SettingsButton';
 import ProfileAvatar from './app/Components/ProfileAvatar/ProfileAvatar';
 import Post from './app/Components/Post/Post';
 import Swiper from 'react-native-swiper';
-import { Divider } from 'react-native-paper';
+import {Divider} from 'react-native-paper';
+import firestore from '@react-native-firebase/firestore';
 import * as ScreenshotDetector from 'react-native-screenshot-detect';
 
-
 export default class FeedScreen extends React.Component {
   constructor(props) {
     super(props);
     this.state = {
       swiperIndex: 0,
+      posts: [],
+      loading: false,
     };
   }
 
   componentDidMount() {
+    this.fetchPosts();
+
     this.eventEmitter = ScreenshotDetector.subscribe(() => {
-      Alert.alert('The post with caption', posts[this.state.swiperIndex].caption, 'was screenshotted');
+      Alert.alert(
+        'The post with caption',
+        posts[this.state.swiperIndex].caption,
+        'was screenshotted',
+      );
     });
   }
 
   componentWillUnmount() {
     ScreenshotDetector.unsubscribe(this.eventEmitter);
   }
 
+  fetchPosts = () => {
+    const subscriber = firestore()
+      .collection('posts')
+      .onSnapshot(querySnapshot => {
+        const posts = [];
+
+        querySnapshot.forEach(documentSnapshot => {
+          posts.push({
+            ...documentSnapshot.data(),
+            key: documentSnapshot.id,
+          });
+        });
```

Comment:
Make sure to include catch(....) statements in the case that errors are encountered